Author: SZTAKI developers
Description: Files uploaded shall also be writable by the group
Disclaimer: This patch aims at being adopted by upstream. It may
 however be Debian-specific. Here, users are commonly created with
 a group of the same name.

Index: boinc/sched/file_upload_handler.cpp
===================================================================
--- boinc.orig/sched/file_upload_handler.cpp	2011-10-08 14:11:57.000000000 +0200
+++ boinc/sched/file_upload_handler.cpp	2011-10-08 14:32:58.000000000 +0200
@@ -119,7 +119,7 @@
     //
     int fd = open(path,
         O_WRONLY|O_CREAT,
-        S_IRUSR|S_IWUSR|S_IRGRP|S_IROTH
+        S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH
     );
     if (fd<0) {
         return return_error(ERR_TRANSIENT,
